onnxruntime/core/platform/windows/env.cc

```diff
@@ -471,7 +494,11 @@ class WindowsEnv : public Env {
   }
 
   virtual Status LoadDynamicLibrary(const std::string& library_filename, void** handle) const override {
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
+    *handle = ::LoadPackagedLibrary(ToWideString(library_filename).c_str(), 0);
```

Comment:
When calling the UWP build dll in a desktop application, now we have different behavior where dlls can potentially be loaded from the application directory rather than the module directory. (with a different search order)

For UWP apps this is probably okay as the search order should be controlled by the sandbox.

Reply:
But we'll have a separate build for Store app and desktop. The Store build won't run as an uncontainerized desktop app.

Reply:
There's an `#else` branch for the non-app case, that calls LoadLibraryExA

Reply:
WinUI only creates a Store build, and then relies on the vcforwarders to have the store binaries run everywhere.

If possible we should ensure that we follow a similar principle.

Reply:
There isn't a common set of APIs that goes from Desktop to UWP to load libraries during runtime. I think that would only be possible if we delayload LoadPackagedLibrary/LoadLibraryExA depending on the platform (and CreateFile2/CreateFileW, GetModuleHandle, SetThreadDescription...).

I didn't know about vcforwarders, are you talking about this project: https://github.com/microsoft/vcrt-forwarders ? Do we want to support the use case of running a Store build on desktop (or vice-versa) and just work?